packages/labs/src/components/select/multiSelect.tsx

```diff
@@ -0,0 +1,256 @@
+    selectedItems?: T[];
+
+    /**
+     * Custom renderer for an item in the dropdown list. Receives a boolean indicating whether
+     * this item is active (selected by keyboard arrows) and an `onClick` event handler that
+     * should be attached to the returned element.
+     */
+    itemRenderer: (itemProps: ISelectItemRendererProps<T>) => JSX.Element;
+
+    /** React child to render when filtering items returns zero results. */
+    noResults?: string | JSX.Element;
+
+    /**
+     * Whether the popover opens on key down or when `TagInput` is focused.
+     * @default false
+     */
+    openOnKeyDown?: boolean;
+
+    /** Props to spread to `Popover`. Note that `content` cannot be changed. */
+    popoverProps?: Partial<IPopoverProps>;
+
+    /**
+     * Whether the filtering state should be reset to initial when an item is selected
+     * (immediately before `onItemSelect` is invoked), or when the popover closes.
+     * The query will become the empty string and the first item will be made active.
+     * @default true
+     */
+    resetOnSelect?: boolean;
+
+    /** Props to spread to `TagInput`. */
+    tagInputProps?: Partial<ITagInputProps>;
+
+    /** Custom renderer to transform an item into a string for tags. */
+    tagRenderer: (item: T) => string;
+}
+
+export interface IMultiSelectState<T> {
+    activeItem?: T;
+    isOpen?: boolean;
+    query?: string;
+}
+
+@PureRender
+export class MultiSelect<T> extends AbstractComponent<IMultiSelectProps<T>, IMultiSelectState<T>> {
+    public static displayName = "Blueprint.MultiSelect";
+
+    public static ofType<T>() {
+        return MultiSelect as new () => MultiSelect<T>;
+    }
+
+    public state: IMultiSelectState<T> = {
+        isOpen: false,
+        query: "",
+    };
+
+    private TypedQueryList = QueryList.ofType<T>();
+    private input: HTMLInputElement;
+    private queryList: QueryList<T>;
+    private queryListKeyDown: React.EventHandler<React.KeyboardEvent<HTMLElement>>;
+    private refHandlers = {
+        input: (ref: HTMLInputElement) => this.input = ref,
+        queryList: (ref: QueryList<T>) => this.queryList = ref,
+    };
+
+    public render() {
+        // omit props specific to this component, spread the rest.
+        const {
+            itemRenderer,
+            noResults,
+            openOnKeyDown,
+            popoverProps,
+            resetOnSelect,
+            tagInputProps,
+            ...props,
+        } = this.props;
+
+        return <this.TypedQueryList
+            {...props}
+            activeItem={this.state.activeItem}
+            onActiveItemChange={this.handleActiveItemChange}
+            onItemSelect={this.handleItemSelect}
+            query={this.state.query}
+            ref={this.refHandlers.queryList}
+            renderer={this.renderQueryList}
+        />;
+    }
+
+    private renderQueryList = (listProps: IQueryListRendererProps<T>) => {
+        const { tagInputProps = {}, popoverProps = {} } = this.props;
```

Comment:
Yeah that's what I wanted, but hit a road block. We went over this earlier in this thread. That's the problem I'm hitting:
![image](https://user-images.githubusercontent.com/199754/27563452-9a5b32cc-5a86-11e7-8b1b-127ba47e57f9.png)

I also want to be able to set default props for `openOnKeyDown` and `resetOnSelect`